overlord/ifacestate/handlers.go

```diff
@@ -328,6 +325,41 @@ func (m *InterfaceManager) undoDiscardConns(task *state.Task, _ *tomb.Tomb) erro
 	return nil
 }
 
+func getDynamicHookAttributes(task *state.Task) (map[string]interface{}, map[string]interface{}, error) {
+	var plugAttrs, slotAttrs map[string]interface{}
+
+	if err := task.Get("plug-dynamic", &plugAttrs); err != nil {
+		return nil, nil, err
```

Comment:
should we ignore NoState errors to be more backward compatible?

Reply:
Done.